### ISSUE TITLE
CheckComboBox mvvm problem

### ISSUE THREAD

**Issue by xceedsoftware** [xceedsoftware/wpftoolkit#753]
<b>md110[CodePlex]</b> <br />hi

this is my sample

when checked multiple item in CheckComboBox

but only one checked item CheckComboBox save to database

how to save All checked items CheckComboBox to database?

thanks

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />hi 

yes, ABC and DEF are Keys from my Document or Person classes


and also I changed this metod: 

__this.HasMany( t = t.Persons )
.WithMany( p = p.Documents )
.Map( m = m.MapLeftKey(PersonID).MapRightKey(DocID) );__


Or 

_this.HasMany( t = t.Persons )
.WithMany( p = p.Documents )
.Map( m = m.MapLeftKey(DocID).MapRightKey( PersonID) );__



but exception error: 

An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while
 saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details


I believe my problem are other solution 

thank you

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

__this.HasMany( t = t.Persons )
.WithMany( p = p.Documents )
.Map( m = m.MapLeftKey(ABC).MapRightKey(DEF) );__ 

has to be redone to respond to your needs, whatever relashionship you need (Many to Many, Many to One...). ABC and DEF should be Keys from your Document or Person classes.

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />Hi
thank you 
You are right this is out of your scope. 


__this.HasMany( t = t.Persons )
.WithMany( p = p.Documents )
.Map( m = m.MapLeftKey(ABC).MapRightKey(DEF) );__ 

used for Many To Many Relationship 

but not work for me
when use this Code Exception: 

An error occurred while saving entities that do not expose foreign key properties for their relationships. The EntityEntries property will return null because a single entity cannot be identified as the source of the exception. Handling of exceptions while
 saving can be made easier by exposing foreign key properties in your entity types. See the InnerException for details

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

You are right, I commented the lines of code to access the database since I don't have access to it. From what I see, it is a relationship error in the constructor of DocumentMap(). You should try to rework the relationship you want, considering the PropertyType
 change (Person to ObservableCollectionPerson in Document). 

Maybe something like 
this.HasMany( t = t.Persons )
.WithMany( p = p.Documents )
.Map( m = m.MapLeftKey(ABC).MapRightKey(DEF) ); 

Bu this is out of our scope. We focus on the toolkit, not a database or the way it is constructed.

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />hi 

because i am using Entity Framework and database in this project 

example :
  public ObservableCollectionPerson PersonListCollection
        {
            get
            {
              context.Persons.Load();
              return context.Persons.Local;

            }
        }
when call context.Persons.Load(); exception Error in DocumentMap class


// Relationships
        this.HasOptional(t = t.Persons[0])
            .WithMany(t = t.Documents)
            .HasForeignKey(d = d.PersonID);

Error is:
The expression 't = t.Persons.get_Item(0)' is not a valid property expression. The expression should represent a property: C#: 't = t.MyProperty'  VB.Net: 'Function(t) t.MyProperty
Do you use DataBase? or use context.Persons.Load(); in PersonListCollection? 

you've use
 public ObservableCollectionPerson PersonListCollection
        {
            get
            {
             // context.Persons.Load();
              //return context.Persons.Local;
              return m_personList;               
            }
        }

you are comment this line: // context.Persons.Load(); and Disabled ef
For this reason, work for you 

if use context.Persons.Load(); 

To give you an error

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

What exactly is not working ? I can compile the sample from the link you provided and I can click on buttons and select items from the CheckComboBox without any exceptions or crashes.

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />hi
Not working for me
please edit last that sample and submit for me
TEXT


thanks

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />The expression 't = t.Persons.get_Item(0)' is not a valid property expression. The expression should represent a property: C#: 't = t.MyProperty' VB.Net: 'Function(t) t.MyProperty'.

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />Hi, 

Thank you for your
I Replaced public virtual Person Person { get; set; } 

with
public virtual ObservableCollectionPerson Persons { get; set; }
but not work And Exception:
The expression 't = t.Personss.get_Item(0)' is not a valid property expression. The expression should represent a property: C#: 't = t.MyProperty'  VB.Net: 'Function(t) t.MyProperty'.

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

You also have to modify the Document class :
Replace 
public virtual Person Person { get; set; }
with
public virtual ObservableCollectionPerson Persons { get; set; }

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />Hi 

Not Work 

Documentdata.Persons = SelectedItemPersons; 

Documentdata has not Persons and not access to Persons 

and
this.HasOptional(t = t.Persons[0])
            .WithMany(t = t.Documents)
            .HasForeignKey(d = d.PersonID);

has bug and not work beacause not access Persons[0]

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

I don't have access to the Database, but here's a test that could work.
 private void DoAddNew()
        {
            Documentdata = Documentdata ?? new Document();
            Documentdata.Persons = SelectedItemPersons;

            RaisePropertyChanged(Persondata);
            RaisePropertyChanged(Documentdata);
        }
In DocumentMap, I'm not sure about this change :
  // Relationships
this.HasOptional(t = t.Persons[0])
                .WithMany(t = t.Documents)
                .HasForeignKey(d = d.PersonID);
You could have to modify some stuff to match your needs, but at least, this compiles.
Basically, the Documentdata holds a collection of Person instead of only 1 Person.

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />hi 

i am using SelectedItemsOverride={Binding SelectedItemPersons} in xaml


my problem is in View model 

i Do not know, how to use SelectedItemsOverride and values of SelectedItemsOverride in
viewmodel and save to database? 

please help me? 

thanks

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

SelectedItemPersons is a ObservableCollectionPerson while Documentdata.Person is a Person, so the conversion can't be done. This is the reason of the exception.


Can Documentdata.Person be changed for Documentdata.Persons, a ObservableCollectionPerson ? If Yes, removing

SelectedValue={Binding Documentdata.PersonID,Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}
and using 
SelectedItemsOverride={Binding SelectedItemPersons}
should do it. 

If not, Documentdata.Person can only set 1 Person. So you won't be able to save more than 1 Person.

**Comment by xceedsoftware** [comment]
<b>md110[CodePlex]</b> <br />hi
Thank you for your attention
All Right , Thse 3 Properties shouldn't be used together
Unused the SelectedMemberPath And SelectedItem in this sample
i forget Remove this tags from Xaml code 

i am using SelectedValue And binding To Documentdata.PersonID and save PersonID value to Document Table

but only one item checked save. 

i am understand use SelectedItemsOverride={Binding SelectedItemPersons} but i Do not know
how to use SelectedItemPersons in view model and save values to database 

when

private void DoAddNew()
    {
        Documentdata = Documentdata ?? new Document();
        Documentdata.Person = SelectedItemPersons ;
        RaisePropertyChanged(Persondata);
        RaisePropertyChanged(Documentdata);
    }


in line 2 exeption error: 

cannot convert type observablecollection to target type ............................


please edit my sample and help me to resolve this problem And submit 

thank you

**Comment by xceedsoftware** [comment]
<b>BoucherS[CodePlex]</b> <br />Hi, 

I'm not sure I understand the sample. On the CheckComboBox, you set the SelectedMemberPath, the SelectedValue and the SelectedItem, but all these properties try to set the current selected item.


-SelectedMemberPath : using Binding SelectedItemPerson says that he checkComboBox.SelectedItems will be the Persons with the path SelectedItemPerson == true. But SelectedItemPerson is not a path of a Person.


-SelectedValue : using Binding Documentdata.PersonID says that the checkComboBox.SelectedItems will be the Persons with PersonID specified in Documentdata.PersonID. Since DocumentData.PersonID is a int, only 1 Person can be selected in the CheckComboBox.


-SelectedItem : using {Binding SelectedItemPerson} says that checkComboBox.SelectedItems will be SelectedItemPerson : only this Person.


Thse 3 Properties shouldn't be used together. Doc : 
https://wpftoolkit.codeplex.com/wikipage?title=CheckComboBox=Documentation


From what I understand, PersonViewModel.DoSave() method will add Documentdata (which is a Document) to context.Documents and save the changes to the database. But there is only 1 Document being added to context.Documents and a Document only points to 1 Person.
 I don't see how persons could be saved. 

If you have a property like :
 public ObservableCollectionPerson _SelectedItemPersons = new ObservableCollectionPerson();
        public ObservableCollectionPerson SelectedItemPersons
        {
          get
          {
            return _SelectedItemPersons;
          }
          set
          {
            if( _SelectedItemPersons != value )
            {
              _SelectedItemPersons = value;
              RaisePropertyChanged( SelectedItemPersons );
            }
          }
        }
and the CheckComboBox have 
SelectedItemsOverride={Binding SelectedItemPersons}
then you could save the property SelectedItemPersons.
